### PR TITLE
Add KPI dataset.

### PR DESCRIPTION
**Pull request by vitillo**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1253644.

### CODE REVIEW

File: src/main/scala/views/KPI.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import telemetry.spark.sql.aggregates._
+
+class Conf(args: Array[String]) extends ScallopConf(args) {
+  val from = opt[String]("from", descr = "From submission date", required = false)
+  val to = opt[String]("to", descr = "To submission date", required = false)
+  verify()
+}
+
+object KPIView {
+  private val hll = new HyperLogLog(10)
```
Comment (mreid-moz) [inline src/main/scala/views/KPI.scala]: why 10?

Comment (vitillo) [inline src/main/scala/views/KPI.scala]: I have been using a low value just for testing purposes.

File: src/main/scala/views/KPI.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.rogach.scallop._
+import telemetry.spark.sql.aggregates._
+
+class Conf(args: Array[String]) extends ScallopConf(args) {
+  val from = opt[String]("from", descr = "From submission date", required = false)
+  val to = opt[String]("to", descr = "To submission date", required = false)
+  verify()
+}
+
+object KPIView {
```
Comment (mreid-moz) [inline src/main/scala/views/KPI.scala]: I don't think `KPI` is a very informative name, I would prefer something like "ClientCountView"

File: src/main/scala/views/KPI.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    val hadoopConf = sc.hadoopConfiguration
+    hadoopConf.set("fs.s3n.impl", "org.apache.hadoop.fs.s3native.NativeS3FileSystem")
+
+    val df = sqlContext.read.load("s3://telemetry-parquet/churn/v1")
+    val subset = df.where(s"submission_date_s3 >= $from and submission_date_s3 <= $to")
+    val aggregates = aggregate(subset)
+
+    val appConf = ConfigFactory.load()
+    val parquetBucket = appConf.getString("app.parquetBucket")
+    aggregates.write.parquet(s"s3://$parquetBucket/KPI/v$from$to")
```
Comment (mreid-moz) [inline src/main/scala/views/KPI.scala]: Is Submission Date one of the dimensions we'll be using? If so, maybe we should use a path like `s3://bucket/viewname/vX/submission_date_s3=YYYYmmdd/<files>`

Then we can incrementally update the aggregates each day.

File: src/main/scala/views/KPI.scala (hunk 1 above)
Comment (mreid-moz) [inline src/main/scala/views/KPI.scala]: We should add a comment in the source that either links to the table [here](https://github.com/twitter/algebird/blob/0.10.0/algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala#L194-L210) or just notes that the expected error rate using 10 bits is ~3%.

File: src/test/scala/KPITest.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    import sqlContext.implicits._
+
+    val dataset = sc.parallelize(Submission.randomList).toDF()
+    val aggregates = KPIView.aggregate(dataset)
+
+    val dimensions = Set(KPIView.dimensions:_*) -- Set("clientId", "hll")
+    (Set(aggregates.columns:_*) -- Set("clientId", "hll")) should be (dimensions)
+
+    val estimates = aggregates.select(hll_cardinality(col("hll"))).collect()
+    estimates.foreach(x => x(0) should be (Submission.dimensions("clientId").size))
```
Comment (mreid-moz) [inline src/test/scala/KPITest.scala]: If I understand correctly, this is checking that each combination of dimensions ends up containing the expected count.

Could we also add a test that combines across several dimensions and see if that also contains the expected count? For example, roll together all the `hll`s where `normalizedChannel` is `release` and ensure that it still matches `Submission.dimensions("clientId").size`

File: src/main/scala/views/KPI.scala (hunk 3 above)
Comment (vitillo) [inline src/main/scala/views/KPI.scala]: I am planning to regenerate the whole report periodically.

**Pull request merged**